opm/parser/eclipse/EclipseState/Schedule/Compsegs.cpp

```diff
@@ -0,0 +1,206 @@
+                       WellCompletion::DirectionEnum dir_in, double center_depth_in, int segment_number_in)
+    : m_i(i_in),
+      m_j(j_in),
+      m_k(k_in),
+      m_branch_number(branch_number_in),
+      m_distance_start(distance_start_in),
+      m_distance_end(distance_end_in),
+      m_dir(dir_in),
+      m_center_depth(center_depth_in),
+      m_segment_number(segment_number_in)
+    {
+    }
+
+    std::vector<CompsegsPtr> Compsegs::compsegsFromCOMPSEGSKeyword(DeckKeywordConstPtr compsegsKeyword,
+                                                                   EclipseGridConstPtr grid) {
+        // the thickness of grid cells will be required in the future for more complete support.
+        // Silence warning about unused argument
+        static_cast<void>(grid);
+
+        // only handle the second record here
+        // The first record here only contains the well name
+        std::vector<CompsegsPtr> compsegs;
+
+        for (size_t recordIndex = 1; recordIndex < compsegsKeyword->size(); ++recordIndex) {
+            DeckRecordConstPtr record = compsegsKeyword->getRecord(recordIndex);
+            // following the coordinate rule for completions
+            const int I = record->getItem("I")->getInt(0) - 1;
+            const int J = record->getItem("J")->getInt(0) - 1;
+            const int K = record->getItem("K")->getInt(0) - 1;
+            const int branch = record->getItem("BRANCH")->getInt(0);
+
+            double distance_start;
+            double distance_end;
+            if (record->getItem("DISTANCE_START")->hasValue(0)) {
+                distance_start = record->getItem("DISTANCE_START")->getSIDouble(0);
+            } else if (recordIndex == 1) {
+                distance_start = 0.;
+            } else {
+                // TODO: the end of the previous connection or range
+                // 'previous' should be in term of the input order
+                // since basically no specific order for the completions
+                throw std::runtime_error("this way to obtain DISTANCE_START not implemented yet!");
+            }
+            if (record->getItem("DISTANCE_END")->hasValue(0)) {
+                distance_end = record->getItem("DISTANCE_END")->getSIDouble(0);
+            } else {
+                // TODO: the distance_start plus the thickness of the grid block
+                throw std::runtime_error("this way to obtain DISTANCE_END not implemented yet!");
+            }
+
+            WellCompletion::DirectionEnum direction;
+            if (record->getItem("DIRECTION")->hasValue(0)) {
+                direction = WellCompletion::DirectionEnumFromString(record->getItem("DIRECTION")->getString(0));
+            } else if (!record->getItem("DISTANCE_END")->hasValue(0)) {
+                throw std::runtime_error("the direction has to be specified when DISTANCE_END in the record is not specified");
+            }
+
+            int end_IJK;
```

Comment:
No need to properly internalize the `END_IJK`value; just check if it is set or not:
```C++
if (!record->getItem<ParserKeywords::COMPSEGS::END_IJK>()->hasValue(0)) {
    CompsegsPtr new_compsegs = std::make_shared<Compsegs>(I, J, K, branch, distance_start, distance_end,
                                                                                                         direction, center_depth, segment_number);
    compsegs.push_back(new_compsegs);
} else { // a range is defined. genrate a range of Compsegs
                throw std::runtime_error("entering COMPSEGS entries with a range is not supported yet!");
}
```